app/lib/features/onboarding/pages/analytics_opt_in_page.dart

```diff
@@ -26,61 +34,280 @@ class AnalyticsOptInPage extends ConsumerWidget {
+    final matomoAnalyticsEnabled =
+        ref.watch(allowMatomoAnalyticsProvider).valueOrNull ?? false;
 
-    return ListTile(
-      onTap: () => toggle(null),
-      title: Text(lang.sendCrashReportsTitle),
-      subtitle: Text(lang.sendCrashReportsInfo),
-      leading: Switch(
-        value: allowReportSending,
-        onChanged: (value) => toggle(value),
-      ),
+    return Column(
+      children: [
+        FutureBuilder<bool>(
+          future: Future.wait([
+            _getAnalyticsPreference(basicTelemetryPref),
+            _getAnalyticsPreference(researchPref),
+          ]).then(
+            (values) =>
+                values.every((value) => value) &&
+                allowReportSending &&
+                matomoAnalyticsEnabled,
+          ),
+          builder: (context, snapshot) {
+            final allEnabled = snapshot.data ?? isNightly;
+            return _buildSwitchOption(lang.togglleAll, allEnabled, () async {
+              final newValue = !allEnabled;
+              await setCanReportToSentry(newValue);
+              await setMatomoAnalyticsEnabled(newValue);
+              await _setAnalyticsPreference(basicTelemetryPref, newValue);
+              await _setAnalyticsPreference(researchPref, newValue);
+              ref.invalidate(allowSentryReportingProvider);
+              ref.invalidate(allowMatomoAnalyticsProvider);
+            }, context);
+          },
+        ),
+        _buildCrashAnalyticsCard(
+          lang.sendCrashReportsTitle,
+          lang.sendCrashReportsInfo,
+          allowReportSending,
+          (value) async {
+            await setCanReportToSentry(value);
+            ref.invalidate(allowSentryReportingProvider);
+          },
+          context,
+          textTheme,
+        ),
+        _buildAnalyticsCard(
+          lang.basicTelemetry,
+          lang.basicTelemetryInfo,
+          basicTelemetryPref,
+          context,
+          textTheme,
+          ref,
+        ),
+        _buildCrashAnalyticsCard(
+          lang.appAnalytics,
+          lang.appAnalyticsInfo,
+          matomoAnalyticsEnabled,
+          (value) async {
+            await setMatomoAnalyticsEnabled(value);
+            ref.invalidate(allowMatomoAnalyticsProvider);
+          },
+          context,
+          textTheme,
+        ),
+        _buildAnalyticsCard(
+          lang.research,
+          lang.researchInfo,
+          researchPref,
+          context,
+          textTheme,
+          ref,
+        ),
+      ],
     );
   }
 
-  Widget _buildSkipActionButton(BuildContext context) {
-    return ActerPrimaryActionButton(
-      key: AnalyticsOptInPage.skipBtn,
-      onPressed: () => context.goNamed(Routes.main.name),
-      child: Text(
-        L10n.of(context).done,
-        style: Theme.of(context).textTheme.bodyMedium,
+  // Reusable widget for switch toggle option without card
+  Widget _buildSwitchOption(
+    String text,
+    bool value,
+    Future<void> Function() onToggle,
```

Comment:
`Future<void> Function() onToggle`

Why it is without new value in function arg?

Reply:
This is still not addressed. 
```suggestion
        Future<void> Function(bool) onToggle,
```